src/Middlewares/AuthMiddleware.php

```diff
@@ -67,6 +70,7 @@ public function __invoke($request, $response, $next)
                 $user = $this->isAuthorised($authUser, $authPass);
 
                 if (!$user) {
+                    $this->grav->fireEvent(Constants::EVENT_ON_API_UNAUTHORIZED_REQUEST, new Event(['user' => $sessionUser, 'roles' => $this->roles]));
```

Comment:
Hmm. This shouldn't be `$sessionUser` as it doesn't exist in this case... But `$user` also wouldn't exist... What should we pass here?

Reply:
I am actually not convinced that this is a necessary event to emit as I cannot see any usecases aside from access logging (which should be handled elsewhere). I said the same about the get-endpoints but after some thinking I do see a case: tracking a recently viewed or something similar which could support a recommender system (would need to be extended with user information though). For unauthorized requests, I don't see such a case. So I would actually propose removing this event again. Should there be a usecase later on, it can always be added. What do you think?

Reply:
Yeah, other than perhaps some alerting/monitoring integration where maybe you care about the number of failed requests, I'm also not sure what this event would be useful for.

I think let's just remove it for now. As you say, we can easily add it later if there's a legitimate need for it.